### PR TITLE
Docs: Migrate special_methods.rst to pure python

### PR DESCRIPTION
**Pull request by matusvalo**
Part of #4187.

### CODE REVIEW

**Comment by matusvalo** [comment]
The only thing missing are unction definitions in tables - https://github.com/cython/cython/blob/3d4f7a1978f4c03a941a2dcff8136b10638476cd/docs/src/userguide/special_methods.rst#sequences-and-mappings

Not sure how to migrate it - adding button for table I consider is ugly :-(

**Comment by da-woods** [comment]
What needs to be changed with the tables? They don't look too specific to any particular syntax to me?

**Comment by matusvalo** [comment]
> What needs to be changed with the tables? They don't look too specific to any particular syntax to me?

I am referring to this conversation: https://github.com/cython/cython/pull/4508#discussion_r772311765 and following:

image

This syntax is more based on Cython language rather than Python annotation.

**Comment by scoder** [comment]
> image
> 
> This syntax is more based on Cython language rather than Python annotation.

If it's really just about these few, then I'd say, let's assume that users who want to implement the buffer interface are somewhat familiar with C and its type syntax. I think we can keep these as they are.

**Comment by da-woods** [comment]
Just closing and re-opening to re-check the test status.

File: docs/examples/userguide/special_methods/total_ordering.pyx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+ 
+@cython.total_ordering
```
[the remark this answers is not in the document]
Comment (matusvalo) [reply docs/examples/userguide/special_methods/total_ordering.pyx:3]: > Tho `(c)import cython` [should be automatically included](https://github.com/cython/cython/pull/4226#issuecomment-860553277). And all of the previous related PRs didn't include it either. It's kind of a convention.

Yes, but on the other hand it fixed failing build - not sure why

[the remark this answers is not in the document]
Comment (da-woods) [reply docs/examples/userguide/special_methods/total_ordering.pyx:3]: https://github.com/cython/cython/blob/579fcbc29bbc097af323a504447bebf6e2696231/runtests.py#L856

I think it's only applied to .py files

[the remark this answers is not in the document]
Comment (matusvalo) [reply docs/examples/userguide/special_methods/total_ordering.pyx:3]: > I think it's only applied to .py files

Relaxing this condition to other than .py files is breaking the suite. Can we go with the PR as it is?

**Comment by da-woods** [comment]
Thanks

**Comment by matusvalo** [comment]
Thank you @da-woods!

**Pull request merged**